cpg_infra/github_wif/driver.py

```
@@ -0,0 +1,436 @@
+    )
+
+    return gcp.serviceaccount.IAMMember(
+        resource_key,
+        service_account_id=service_account.name,
+        role='roles/iam.workloadIdentityUser',
+        member=principal,
+    )
+
+
+def grant_artifact_registry_access(
+    resource_key: str,
+    service_account: gcp.serviceaccount.Account,
+    project_id: str,
+    location: str,
+    registry_name: str,
+    access: Literal['read', 'write'],
+) -> gcp.artifactregistry.RepositoryIamMember:
+    """
+    Grant Artifact Registry access to a service account.
+
+    Args:
+        resource_key: Unique resource key for Pulumi
+        service_account: Service account to grant access to
+        project_id: GCP project ID
+        location: Registry location
+        registry_name: Name of the artifact registry
+        access: Type of access ('read' or 'write')
+
+    Returns:
+        IAM member binding
+    """
+    role = (
+        'roles/artifactregistry.writer'
+        if access == 'write'
+        else 'roles/artifactregistry.reader'
+    )
+
+    member = pulumi.Output.concat('serviceAccount:', service_account.email)
+
+    return gcp.artifactregistry.RepositoryIamMember(
+        resource_key,
+        project=project_id,
+        location=location,
+        repository=registry_name,
+        role=role,
+        member=member,
+    )
+
+
+def manage_github_secrets(
+    github_repo: str,
+    environment: str,
+    provider_name: pulumi.Output[str],
+    service_account_email: pulumi.Output[str],
+) -> None:
+    """
+    Create GitHub secrets for WIF in the specified repository environment.
+
+    Args:
+        github_repo: Full GitHub repo path (org/repo)
+        environment: Environment name
+        provider_name: WIF provider resource name
+        service_account_email: Service account email
+    """
+    # Parse org and repo
+    if '/' not in github_repo:
+        # Should be caught by validation, but safe fallback
+        return
```

raise an error here rather than failing silently